### ISSUE TITLE
Duplicate tabs

### ISSUE THREAD

**Issue by Ben3eeE** [atom/atom#11391]
### Prerequisites

* [X] Can you reproduce the problem in [safe mode](http://flight-manual.atom.io/hacking-atom/sections/debugging/#check-if-the-problem-shows-up-in-safe-mode)?
* [X] Are you running the [latest version of Atom](http://flight-manual.atom.io/hacking-atom/sections/debugging/#update-to-the-latest-version)?
* [X] Did you check the [debugging guide](flight-manual.atom.io/hacking-atom/sections/debugging/)?
* [X] Did you check the [FAQs on Discuss](https://discuss.atom.io/c/faq)?
* [X] Are you reporting to the [correct repository](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#atom-and-packages)?
* [X] Did you [perform a cursory search](https://github.com/issues?q=is%3Aissue+user%3Aatom+-repo%3Aatom%2Felectron) to see if your bug or enhancement is already reported?

For more information on how to write a good [bug report](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#how-do-i-submit-a-good-bug-report) or [enhancement request](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#how-do-i-submit-a-good-enhancement-suggestion), see the `CONTRIBUTING` guide.

### Description

This is very likely a very low priority issue caused by a race condition in the deduper (@lee-dohm  theory) but it is possible (if you are fast enough) to open the same tab multiple times (I managed twice at least).

### Steps to Reproduce

1. Open tab from tree view by clicking on it.
2. Immediately after step 1 press enter.
3. Observe you have 2 tabs of the same file open, in the case where pending pane items option is true you have the one pending tab and one confirmed tab, without pending pane items option you have 2 confirmed tabs.

For me this was easier at my work computer and with the first tab that i open in that atom window, as the first tab takes several seconds to load you have a bigger window to execute the 2nd step.
At home I can reproduce it but I have to try for a long time.

**Expected behavior:** Atom to switch to already open tab at all times.

**Actual behavior:** Atom opens multiple (at least 2 confirmed) tabs of the same file.

### Versions
1.6.X
Windows 7.

**Comment by karloluis** [comment]
If I do both quick enough I can reproduce this easily. Not a big problem in my case since I don't tend to use such behavior. Still a bug nonetheless.

**Comment by albinekb** [comment]
Reproduced, however not a big problem

**Comment by Ben3eeE** [comment]
Since 1.7.0 this has become more of a problem for me. I can now reproduce it by just double clicking files in the tree view, every time I get a small lag when opening I have double tabs open.
Before I had to use the weird steps of pressing enter after clicking on the file which I never did.
I use pending pane items.

**Comment by albertosantini** [comment]
I noticed it too, since Atom 1.7.0 (on Windows 8 and Windoww 10).

After starting Atom, without any file opened, I can reproduce it for the first file I open: double clicking a file in the tree view with `Allow pending Pane Items` enabled.

Two tabs are opened: a normal one and another pending.
Indeed if I click on another file, the second tab is replaced from the last file clicked.

**Comment by PlaidPhantom** [comment]
I think I'm seeing the same behavior as @albertosantini.  If I click a file in the Tree View, let the Pending Pane open, and then double-click, I get the expected behavior.  If I immediately double-click a file, I get the normal pane *and* a Pending Pane with the same file in each.

**Comment by WaxMikhail** [comment]
As of 1.7.4, on Mac OSX, I can consistently reproduce this error. In fact, almost any time I double-click on a file in tree view it opens twice/

**Comment by Ben3eeE** [comment]
@WaxMikhail @kuychaco has merged a PR fixing this issue but it has not been released in any version yet so it is expected that you can reproduce it.
